src/sql/workbench/contrib/notebook/browser/cellViews/markdownToolbar.component.ts

```diff
@@ -258,6 +271,13 @@ export class MarkdownToolbarComponent extends AngularDisposable {
 		} else if (!needsTransform) {
 			if (type === MarkdownButtonType.LINK_PREVIEW) {
 				await insertFormattedMarkdown(linkCalloutResult?.insertEscapedMarkdown, this.getCellEditorControl());
+			} else if (type === MarkdownButtonType.IMAGE_PREVIEW) {
+				if (imageCalloutResult.embedImage) {
+					let base64String = await this.fileToBase64(imageCalloutResult.imagePath, false);
+					let mimeType = await this.fileToBase64(imageCalloutResult.imagePath, true);
+					this.cellModel.addAttachment(mimeType.toString(), base64String.toString(), path.basename(imageCalloutResult.imagePath).replace(' ', '')); await insertFormattedMarkdown(imageCalloutResult?.insertEscapedMarkdown, this.getCellEditorControl());
```

Comment:
imageCalloutResult will always be defined at this point so no need for the `imageCalloutResult?`